src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -1382,4 +1401,21 @@ public void shouldKeepUsingReplicaAfterSettingTimeout() throws SQLException {
         assertThat(connectionProvider.getProvidedConnectionTypes())
             .containsOnly(REPLICA);
     }
+
+    @Test
+    public void shouldResetReadOnlyModeBeforeReleasingConnection() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().build()
+        ).build();
+        connection.setReadOnly(true);
+        connection.prepareStatement(SIMPLE_QUERY).executeQuery();
+        final Connection rawConnection = connectionProvider.singleProvidedConnection();
+        Mockito.reset(rawConnection);
```

Comment:
why do you have to reset mock ?

Reply:
Reset focuses on the `BeforeReleasingConnection` part of the test case